vscode/microsoft-kiota/src/extension.ts

```diff
@@ -706,9 +706,12 @@ async function showUpgradeWarningMessage(clientPath: string, context: vscode.Ext
   }
   const workspaceFileData = await vscode.workspace.fs.readFile(vscode.Uri.file(workspaceFilePath));
   const workspaceFile = JSON.parse(workspaceFileData.toString()) as { kiotaVersion: string };
-  const clientVersion = workspaceFile.kiotaVersion.toLocaleLowerCase();
-  if (clientVersion.toLocaleLowerCase() !== kiotaVersion) {
-    await vscode.window.showWarningMessage(vscode.l10n.t("Client will be upgraded from version {0} to {1}, upgrade your dependencies", clientVersion, kiotaVersion));
+  // don't fail if kiotaVersion isn't in the workspace config file
+  if (workspaceFile.kiotaVersion) {
```

Comment:
It works. we can open it up for review.

Reply:
The problem is that this code path will always be skipped unless I add the `kiotaVersion` to the manifest.

@baywet, does it make sense to split that work to multiple PRs? I was planning on doing all the work in this one.

Reply:
if you feel like the work on complete resolution is going to be extensive, yes, otherwise if you feel like we're talking about a couple of days or less, it's probably not worth the overhead.

Reply:
I'll check this in to unblock the update scenario crash but there will be another PR awaiting resolution of the discussion on #5430